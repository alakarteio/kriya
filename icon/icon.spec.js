/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import Icon from './icon'

const component = ({ ...props }) => {
  return renderer.create(
    <Icon {...props} />,
  )
}


const snap = (props) => {
  const tree = component({ children: 'icon', ...props }).toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Icon', () => {
  describe('graphical (JSX)', () => {
    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should create a primary icon', () => snap({ primary: true }))
    it('should create a secondary icon', () => snap({ secondary: true }))
    it('should create a disabled icon', () => snap({ disabled: true }))
    it('should create an icon without color', () => snap({ noColor: true }))
    it('should create an icon with label to left', () => snap({ labelLeft: 'label' }))
    it('should create an icon with label to right', () => snap({ labelRight: 'label to right' }))
    it('should print labels to both sides', () => snap({ labelLeft: 'left', labelRight: 'right' }))
    it('should take an icon name as children', () => snap({ children: 'different-icon' }))
    it('should have a default behaviour', () => snap({}))
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
