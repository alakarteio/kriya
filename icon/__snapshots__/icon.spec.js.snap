// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Icon graphical (JSX) should add custom className 1`] = `
<div
  className="icon custom colored"
  style={Object {}}
>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should add custom style 1`] = `
<div
  className="icon colored"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create a disabled icon 1`] = `
<div
  className="icon disabled colored"
  style={Object {}}
>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create a primary icon 1`] = `
<div
  className="icon primary colored"
  style={Object {}}
>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create a secondary icon 1`] = `
<div
  className="icon secondary colored"
  style={Object {}}
>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create an icon with label to left 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <label
    htmlFor="mdv-icon"
  >
    label
  </label>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create an icon with label to right 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
  <label
    htmlFor="mdv-icon"
  >
    label to right
  </label>
</div>
`;

exports[`common/Icon graphical (JSX) should create an icon without color 1`] = `
<div
  className="icon"
  style={Object {}}
>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should have a default behaviour 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should print labels to both sides 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <label
    htmlFor="mdv-icon"
  >
    left
  </label>
  <i
    className="mdv-icon"
    name="mdv-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
  <label
    htmlFor="mdv-icon"
  >
    right
  </label>
</div>
`;

exports[`common/Icon graphical (JSX) should take an icon name as children 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="mdv-different-icon"
    name="mdv-different-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;
